core/src/main/java/tachyon/master/BlockInfo.java

```diff
@@ -181,21 +170,52 @@ public synchronized InodeFile getInodeFile() {
           } catch (NumberFormatException nfe) {
             continue;
           }
-          ret.add(new NetAddress(resolvedHost, resolvedPort, -1));
+          ret.add(resolvedHost);
         }
       }
     }
     return ret;
   }
 
+  public synchronized List<NetAddress> getWorkerAddresses() {
```

Comment:
This is not quite the same as `getLocations` since checkpoint locations will not be added. Is that the intention? For how we are using it now I think its okay either way.